### ISSUE TITLE
A component that touches the global system feels like another system

### ISSUE THREAD

**Issue by theronic** [danielsz/system-advanced-example#2]
Hi @danielsz. I opened a [related issue on system](https://github.com/danielsz/system/issues/125) after looking for an example to do reloading of a dynamic handler. On [this line](https://github.com/danielsz/system-advanced-example/blob/master/src/example/handler.clj#L21) a global routes var touches the global `system` var directly to access :db-spec, implying knowledge of the whole system.

This seems to contradict the component philosophy. A function "demands" its dependencies through its arguments. Once you are calling out to a global var, bets are off - you can't unit test that function without monkey-patching system vars.

It feels like the caller should be passing down only instances (ideally, values) of subcomponents that a given function needs to do its work. Maybe I'm doing it wrong.

Could you show an advanced example that passes down conn/db/db-spec while still supporting reloading?

**Comment by danielsz** [comment]
Absolutely, for a clean, correct usage of Component you want this example: https://github.com/danielsz/system-dependency-injection

Direct access of system map is discouraged. There is a section about that in the README: 
https://github.com/danielsz/system#at-runtime-global-system-map-vs-dependency-injection
